swarm/main.cpp

```diff
@@ -103,14 +103,24 @@ init_logging(const bzn::options& options)
 
 
 bool
-init_peers(bzn::bootstrap_peers& peers, const std::string& peers_file, const std::string& peers_url)
+init_peers(bzn::bootstrap_peers& peers, const std::string& peers_file, const std::string& peers_url, const std::string &swarm_info_esr_address, const bzn::uuid_t& swarm_id)
 {
-    if (peers_file.empty() && peers_url.empty())
+    if (peers_file.empty() && peers_url.empty() && swarm_id.empty())
     {
-        LOG(error) << "Bootstrap peers must be specified options (bootstrap_file or bootstrap_url)";
+        LOG(error) << "Bootstrap peers must be specified options (bootstrap_file, bootstrap_url or swarm_id)";
         return false;
     }
 
+    if (!swarm_id.empty())
+    {
+        peers.fetch_peers_from_ESR_contract(swarm_info_esr_address, swarm_id);
+        if (!peers.get_peers().empty())
+        {
+            return true;
+        };
```

Comment:
remove ;

Reply:
Cleaning up. Checking for other instances of that stupidity.

Reply:
ok, maybe not. Too many uses of }; in variable instantiation.. I'll fix the one you pointed out..